src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -133,9 +144,35 @@ public SearchResultPageDTO<Lecture> getAllOnPageWithSize(final SearchTermPageabl
      * @param lecture the lecture to be deleted
      */
     public void delete(Lecture lecture) {
+        if (pyrisWebhookService.isPresent()) {
+            Lecture lectureWithAttachmentUnits = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lecture.getId());
+            List<AttachmentUnit> attachmentUnitList = lectureWithAttachmentUnits.getLectureUnits().stream()
+                    .filter(lectureUnit -> lectureUnit instanceof AttachmentUnit && ((AttachmentUnit) lectureUnit).getAttachment().getLink().endsWith(".pdf"))
+                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).toList();
+            if (!attachmentUnitList.isEmpty()) {
```

Comment:
```suggestion
            if (attachmentUnitList.isPresent()) {
```

Reply:
This is not doing the same thing, I am checking whether the list has elements to send to Pyris and not whether it is optional. Or did I miss something in here ?

Reply:
Yeah, my mistake. I was somehow thinking about an optional.